test-app/tests/integration/components/input-field-test.gts

```diff
@@ -0,0 +1,166 @@
+/* eslint-disable no-undef -- Until https://github.com/ember-cli/eslint-plugin-ember/issues/1747 is resolved... */
+/* eslint-disable simple-import-sort/imports,padding-line-between-statements,decorator-position/decorator-position -- Can't fix these manually, without --fix working in .gts */
+import { fillIn, find, render, setupOnerror } from '@ember/test-helpers';
+import { module, test } from 'qunit';
+
+import InputField from '@crowdstrike/ember-toucan-core/components/form/input-field';
+import { setupRenderingTest } from 'test-app/tests/helpers';
+
+
+module('Integration | Component | InputField', function (hooks) {
+  setupRenderingTest(hooks);
+
+  test('it renders', async function (assert) {
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        data-input
+        />
+    </template>);
+
+    const label = '[data-label]'
+    const input = '[data-input]';
+    const inputId = find(input)?.getAttribute('id') || ''; 
+
+    assert.ok(inputId, 'Expected to have id');
+    assert.dom(label).exists('Expected to have label block rendered');
+    assert.dom(label).hasText('Label', 'Expected to have label text "label"');
+    assert.dom(label).hasAttribute('for', inputId);
+    assert.dom(input).exists('Expected to have input tag rendered');
+    assert.dom(input).hasAttribute('type', 'text');
+    assert.dom(input).hasAttribute('id');
+  });
+
+  test('it encourages @label via assert', async function (assert) {
+    assert.expect(1)
+
+    setupOnerror((e:Error) => {
+      assert.strictEqual(e.message, 'Assertion Failed: input field requires a label', 'Expected an error message if @label is not provided')
+    })
+    await render(<template>
+      {{! @glint-expect-error: should have an error here for missing @label }} 
+      <InputField
+        type="text"
+        />
+    </template>);
+
+  })
+  
+  test('it renders an error', async function (assert) {
+    await render(<template>
+      <InputField
+        @label="Label"
+        type="text"
+        @error="There is an error"
+        data-input
+        />
+    </template>);
+
+    const label = '[data-label]'
+    const input = '[data-input]';
+    const error = '[data-error]';
+  
+    assert.dom(label).exists('Expected to have label component rendered');
+    assert.dom(label).hasText('Label', 'Expected to have label text "Label"');
```

Comment:
Nit: Up to you, but there's a test above validating `label`, so you could probably get away with this test focusing on only testing that the error displays and removing these assertions as they're redundant (same for other tests).